src/Semantics.js

```diff
@@ -196,6 +196,41 @@ Semantics.prototype.checkActionDicts = function() {
   }
 };
 
+Semantics.prototype.toRecipe = function() {
+  var sb = new common.StringBuffer();
+  sb.append('(function() {\n  var grammarFn = ');
+  sb.append(this.grammar.toRecipe());
+  sb.append('  var grammar = grammarFn.call(this);\n');
+  sb.append('  var semantics = grammar.semantics()');
+
+  ['Operation', 'Attribute'].forEach(function(type) {
+    var semantics = this[type.toLowerCase() + 's'];
```

Comment:
I agree, it's cumbersome and confusing to use two different names for things that are so similar. I think right now there are only two differences:

1. Attributes are memoized, and
2. Attributes can't have arguments

...and I could imagine that at some point we would support memoizing with arguments (though it's not clear exactly how it would work).

I'd be in favour of just calling everything "semantic operations", though using "attributes" has the advantage of being connected to attribute grammars.